alphadia/cli.py

```diff
@@ -335,36 +190,38 @@ def run(*args, **kwargs):
         print(f"{alphadia.__version__}")
         return
 
-    config = parse_config(args)
+    user_config = _get_config_from_args(args)
 
-    output_directory = parse_output_directory(args, config)
+    output_directory = _get_from_args_or_config(
+        args, user_config, args_key="output", config_key="output_directory"
+    )
     if output_directory is None:
-        # print help message if no output directory specified
         parser.print_help()
-
         print("No output directory specified.")
         return
 
-    quant_dir = parse_quant_dir(args, config)
-
     reporting.init_logging(output_directory)
-    raw_path_list = parse_raw_path_list(args, config)
-
-    library_path = parse_library(args, config)
-    fasta_path_list = parse_fasta(args, config)
+    logger.progress(f"Saving output to: {output_directory}")
 
+    raw_path_list = _get_raw_path_list_from_args_and_config(args, user_config)
     logger.progress(f"Searching {len(raw_path_list)} files:")
     for f in raw_path_list:
         logger.progress(f"  {os.path.basename(f)}")
 
+    library_path = _get_from_args_or_config(
+        args, user_config, args_key="library", config_key="library"
+    )
     logger.progress(f"Using library: {library_path}")
 
+    fasta_path_list = _get_fasta_list_from_args_and_config(args, user_config)
     logger.progress(f"Using {len(fasta_path_list)} fasta files:")
     for f in fasta_path_list:
         logger.progress(f"  {f}")
 
     # TODO rename all output_directory, output_folder => output_path, quant_dir->quant_path (except cli parameter)
-    logger.progress(f"Saving output to: {output_directory}")
+    quant_dir = _get_from_args_or_config(
+        args, user_config, args_key="quant_dir", config_key="quant_dir"
+    )
```

Comment:
just some reordering to have the extraction of the value and the log statement close by